src/ui/FacetSlider/FacetSlider.ts

```diff
@@ -937,23 +937,45 @@ export class FacetSlider extends Component {
   }
 
   private trySetSliderBoundaryFromState() {
-    const stateValues = this.rangeFromUrlState || this.queryStateModel.get(this.rangeQueryStateAttribute);
-    if (stateValues && stateValues[0] != undefined && stateValues[1] != undefined) {
-      stateValues[0] = Number(stateValues[0]);
-      stateValues[1] = Number(stateValues[1]);
-      this.setupInitialSliderStateStart(stateValues[0]);
-      this.setupInitialSliderStateEnd(stateValues[1]);
-      this.startOfSlider = stateValues[0];
-      this.endOfSlider = stateValues[1];
+    const boundaries = this.getBoundariesFromState();
+    if (!boundaries) {
+      return;
     }
+    this.setupInitialSliderStateStart(boundaries.start);
+    this.setupInitialSliderStateEnd(boundaries.end);
+    this.startOfSlider = boundaries.start;
+    this.endOfSlider = boundaries.end;
   }
 
   private trySetSliderBoundaryFromQueryResult(data: IQuerySuccessEventArgs) {
     const groupByResults = data.results.groupByResults[this.facetQueryController.lastGroupByRequestForFullRangeIndex];
-    if (groupByResults && groupByResults.values.length > 0 && groupByResults.values[0].numberOfResults != 0) {
-      this.setupInitialSliderStateStart(groupByResults.values[0].value.split('..')[0]);
-      this.setupInitialSliderStateEnd(groupByResults.values[groupByResults.values.length - 1].value.split('..')[1]);
+    if (!groupByResults || groupByResults.values.length <= 0 || groupByResults.values[0].numberOfResults == 0) {
+      return;
+    }
+
+    let start = Number(groupByResults.values[0].value.split('..')[0]);
+    let end = Number(groupByResults.values[groupByResults.values.length - 1].value.split('..')[1]);
+
+    const boudariesFromState = this.getBoundariesFromState();
+
+    if (boudariesFromState) {
+      start = Math.min(start, boudariesFromState.start);
+      end = Math.max(end, boudariesFromState.end);
```

Comment:
Could there potentially be a UX issue with this change in that, the difference between `min` and `max` will always either stay the same or increase. After a few searches, could the slider range become too wide/sensitive and no longer useful?

Reply:
> Or does the code only run once on page initialization?

It's exactly like. It's the code that runs only when the slider has not been initialized yet. So, only once per page initialization.

In the case of this specific bugfix, it runs when the first query returns, because the start and end are not specified at config time in the HTML.